### ISSUE TITLE
Odd Permission Issue

### ISSUE THREAD

**Issue by wwilson603** [TheGrandWazoo/freenas-proxmox#43]
Hello,

I am currently experiencing what seems to be a simple permissions issue when trying to access my shares. I have followed this guide here: https://forum.proxmox.com/threads/guide-setup-zfs-over-iscsi-with-pve-5x-and-freenas-11.54611/ to the tee but it still seems that I cannot connect. 

Got through the patches and SSH key creation just fine, but when trying to add the share in Proxmox there is a permissions error:
![2019-07-11 22_12_39-prxmx-vhost-01 - Proxmox Virtual Environment](https://user-images.githubusercontent.com/52811963/61104090-ce1afb80-a429-11e9-8f30-fe5f001b3ab0.png)

I copy-pasted the error message shown in Proxmox into the console and get this:
![2019-07-11 22_13_28-prxmx-vhost-01 - Proxmox Virtual Environment](https://user-images.githubusercontent.com/52811963/61104072-bba0c200-a429-11e9-948b-55ba08a22257.png)

I have tried or verified:
- Manually logging into FreeNAS through the Proxmox console using ssh@[name of freenas server], works just fine using password authentication and I can successfully run the zfs list command,
- That I enabled password authentication and allowed login as root on the FreeNAS end.
- The public key has been uploaded to the FreeNAS server and that it matches the one in Proxmox.

My setup is as follows:
- DAC 10GbE (Intel X520) connection running between the two servers, with a static IP set on each end
- FreeNAS 11.2-U5
- ProxMox 5.4-3 (config for ZFS connection below)
![2019-07-11 22_12_56-prxmx-vhost-01 - Proxmox Virtual Environment](https://user-images.githubusercontent.com/52811963/61104124-ec80f700-a429-11e9-9d3f-5c10c0bd5413.png)

I would imagine I'm just missing something simple here, as I am still new to Linux and making mistakes here and there. Any help would be appreciated. Thanks!

**Comment by raczkow** [comment]
You must login to FreeNAS with SSH key only. Verify your public key location on FreeNAS because it doesn't work correctly.

**Comment by wwilson603** [comment]
Hi raczkow,

I tried running this command from the Proxmox console:
`ssh -i /etc/pve/priv/zfs/192.168.77.2_id_rsa root@192.168.77.2`
and was able to get into FreeNAS using it after punching in my private key passphrase.

I have also verified that the public key is located on the FreeNAS end in the ~/.ssh/authorized_keys directory.

**Comment by wwilson603** [comment]
I figured this out!

When generating the public and private keys, there is an option to put a passphrase on it or **leave it blank for no passphrase.** I opted to put a passphrase on my keys and that was what caused the issue since the command built into the code runs the batchmode flag can't pass the passphrase check.

I appreciate the work you put into the project and I apologize for not recognizing my mistake earlier.
